src/com/t_oster/liblasercut/drivers/GenericGcodeDriver.java

```diff
@@ -483,15 +483,15 @@ protected void line(PrintStream out, double x, double y, double resolution) thro
     String append = "";
     if (nextPower != currentPower)
     {
-      append += String.format(FORMAT_LOCALE, " S%f", nextPower);
+      append += String.format(FORMAT_LOCALE, " S%d", (int) nextPower);
```

Comment:
I think this needs to be reverted, or changed to `%.2f`, as some laser cutters use 0.0-1.0 for their laser power, others use 0-255, etc. The max spindle power setting is configurable in both Grbl and Visicut, so my laser cutter uses 0.0 to 1.0 for its power. So with this setting when I did a 3d laser engrave it all came out as `S0` for every line that wasn't 100% black. Changing it to `%.2f` solves that and gives reasonable accuracy without leading to massive file sizes. You could perhaps vary the formatter based on the user's max spindle power setting (float for < 100, int otherwise), not sure if it is worth the code though?